packages/data-context/src/sources/ProjectDataSource.ts

```diff
@@ -282,13 +282,15 @@ export class ProjectDataSource {
   }
 
   async defaultSpecFileName () {
-    const defaultFileName = 'cypress/e2e/filename.cy.js'
+    const getDefaultFileName = (testingType: TestingType) => `cypress/${testingType}/filename.cy.${this.ctx.lifecycleManager.fileExtensionToUse}`
```

Comment:
This doesn't seem correct. What if `cypress/component` does not exist? Considering the default is something like `src/**/*.cy.{js.jsx}`, it likely won't exist for most users.

Should we make sure the default placeholder is a valid path (or we don't care?)

We could check against the specPattern to make sure it matches (we've already got this logic [here](https://github.com/cypress-io/cypress/blob/10.0-release/packages/graphql/src/schemaTypes/objectTypes/gql-Mutation.ts#L64-L76).

If it does not match, we could do something like:

```js
const root = specPattern.split('/')[0]
const getDefaultFileName = path.join(root, 'filename.cy', this.ctx.lifecycleManager.fileExtensionToUse)`
```

What do you think?

Reply:
That's why it is checked [here](https://github.com/cypress-io/cypress/pull/21411/files/2edb1c396c31c65b00083548a8bc9a319ec1b02c#diff-6085376a4a2ce7de42456fe0abf38e4ddbd45dff4851894f456598e40e586cc7R305-R307) to be sure it's the default specPattern - so, this ^^ is just for the default paths

when there're custom specPatterns as you mention it is updated [here](https://github.com/cypress-io/cypress/pull/21411/files/2edb1c396c31c65b00083548a8bc9a319ec1b02c#diff-6085376a4a2ce7de42456fe0abf38e4ddbd45dff4851894f456598e40e586cc7R309) using the custom pattern and wild cards

Reply:
🎉